massiv/src/Data/Massiv/Core/Index/Ix.hs

```diff
@@ -93,6 +100,9 @@ type family Ix (n :: Nat) = r | r -> n where
 data IxN (n :: Nat) where
   (:>) :: Dimensions (Ix (n - 1)) ~ (n - 1) => {-# UNPACK #-} !Int -> !(Ix (n - 1)) -> IxN n
 
+instance Validity (Ix (n - 1)) => Validity (IxN n) where
+  validate (a :> b) = declare "first element is positive" (a > 0) <> delve "tail" b
```

Comment:
I don't think `Validity` instance for `IxN` should be different for different versions of GHC (I just noticed two instance declarations). Also, correct me if I am wrong, but `IxN` can have first element equal to `0` (eg. `0 :> 0 :. 1`) , since it is not only used for size but for indexing as well. Moreover there, there are places in `Stencil`, as you know, that have negative indices, but I guess as far as `Validity` is concerned it might not be important.

Reply:
> Moreover there, there are places in Stencil, as you know, that have negative indices, but I guess as far as Validity is concerned it might not be important.

If `IxN` with a `0` or a negative value is _ever_ a valid thing, then the validity instance shouldn't try to validate against that.